bsp4kt/src/org/jetbrains/bsp/bsp4kt/Code.kt

```diff
@@ -0,0 +1,24 @@
+package org.jetbrains.bsp.bsp4kt
+
+import bsp4kt.util.untaggedUnionSerializer
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonContentPolymorphicSerializer
+import kotlinx.serialization.json.JsonElement
+
+@Serializable(with = Code.Companion::class)
+sealed interface Code {
+  @Serializable
+  @JvmInline
+  value class StringValue(val value: String): Code {}
+
+  @Serializable
+  @JvmInline
+  value class IntValue(val value: Int): Code {}
+
+  companion object : JsonContentPolymorphicSerializer<Code>(Code::class) {
+    override fun selectDeserializer(element: JsonElement): DeserializationStrategy<Code> {
+      return untaggedUnionSerializer<Code>(stringSerializer = StringValue.serializer(),intSerializer = IntValue.serializer())(element)
```

Comment:
ej ale to w ogóle zakłada że untaggedUnion może zawierać tylko stringi i inty? bo jak masz takie założenie to dało się to prościej zrobić i napisać coś w stylu StringIntUnionSerializer. a jak nie masz takiego założenia no to ta implementacja jest niewystarczająco generyczna

Reply:
no ja założyłam że jak będzie potrzebne inne wsparcie to zostanie dopisane. Ale racja, można już od razu zrobić wsparcie wszystkiego

Reply:
jednak się poddałam z robieniem uniwersalnego serializatora i zrobiłam `StringIntUnionSerializer`, co sądzisz o tym rozwiązaniu?